js/kucoin.js

```diff
@@ -786,6 +786,18 @@ module.exports = class kucoin extends Exchange {
         return (type === 'contract') || (type === 'future') || (type === 'futures'); // * (type === 'futures') deprecated, use (type === 'future')
     }
 
+    isMarginMethod (methodName, params) {
+        //
+        // Helper
+        // @methodName (string): The name of the method
+        // @params (dict): The parameters passed into {methodName}
+        // @return: true if the method used is meant for margin trading, false otherwise
+        //
+        const defaultType = this.safeString2 (this.options, methodName, 'defaultType', 'trade');
+        const tradeType = this.safeString (params, 'tradeType', defaultType);
+        return (tradeType === 'MARGIN_TRADING') || (tradeType === 'margin');
+    }
```

Comment:
Shouldn't we omit 'tradeType' from params? otherwise will be part of the final request

Reply:
tradeType is part of the final request for some methods such as fetchOrderByStatus and cancelAllOrders.
![image](https://user-images.githubusercontent.com/64965782/180895668-7fc8042a-c0d9-4f1b-8ef9-134e8bd0e86f.png)